### ISSUE TITLE
Enhancement of Bank Loan Approval Prediction

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#473]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Enhancement of Bank Loan Approval Prediction</br>
:red_circle: **Aim** : The existing [Bank Loan Approval Prediction](https://github.com/abhisheks008/DL-Simplified/tree/main/Bank%20Loan%20Approval%20Prediction) project is developed using a single neural network model. The aim is to enhance the project by adding new models for this project which will make this project more compact and accurate based on the accuracy scores.</br>
:red_circle: **Dataset** : Mentioned in the dataset folder of the project.</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Vaidehi-Shende** [comment]
Full name : Vaidehi Shende
GitHub Profile Link : https://github.com/Vaidehi-Shende
Email ID : vaidehishende2003@gmail.com
Approach for this Project : To implement various Deep Learning algorithms including CNNs, RNNs, LSTM 
What is your participant role? (Mention the Open Source program):  SWOC

**Comment by abhisheks008** [comment]
Apart from the existing model of this project what are the models you are trying to implement? Can you specify a bit on that.

**Comment by Vaidehi-Shende** [comment]
Enhance the Bank Loan Approval Prediction project by implementing a Multi-Layer Perceptron (MLP), Convolutional Neural Network (CNN), Recurrent Neural Network (RNN) or Long Short-Term Memory (LSTM) and Autoencoders for feature learning after doing Exploratory Data Analysis (EDA), model fitting, hyperparameter tuning, and training/testing, comparing their accuracy scores for optimal model selection.

**Comment by abhisheks008** [comment]
Assigned to @Vaidehi-Shende under SWOC S4.

**Comment by kumar8074** [comment]
Hi @abhisheks008, can you please assign this project to me?

Full name : Lalan Kumar
GitHub Profile Link : https://github.com/kumar8074
Email ID : lalan.k8977@gmail.com
Approach for this Project : I'll first try to use my own DL algorithms and after that i'll utilize the existing algorithms using transfer learning technique and also i'll try out encoders or GANs to get the best accuracy and the evaluation results.
What is your participant role? GSSoC'24

**Comment by abhisheks008** [comment]
You need to enhance the existing project and make the changes accordingly with better results. Are you okay with this?
@kumar8074

**Comment by kumar8074** [comment]
> You need to enhance the existing project and make the changes accordingly with better results. Are you okay with this?
> @kumar8074 

Yes absolutely @abhisheks008

**Comment by abhisheks008** [comment]
Cool. Issue assigned to you @kumar8074

**Comment by kumar8074** [comment]
> Cool. Issue assigned to you @kumar8074 

Thank you @abhisheks008

**Comment by kumar8074** [comment]
Hey @abhisheks008 just had a question, Should I directly complete this project and make a pull request or should i have to show day to day progress? If i have to show day to day progress how to do that ? as i'm quite new to open-source i don't know how to do that can you please help me with that?

**Comment by abhisheks008** [comment]
> Hey @abhisheks008 just had a question, Should I directly complete this project and make a pull request or should i have to show day to day progress? If i have to show day to day progress how to do that ? as i'm quite new to open-source i don't know how to do that can you please help me with that?

Complete your work and then push the code. No need to push a day to day progress. Take your time and come up with the best quality code.

**Comment by kumar8074** [comment]
OK got it

**Comment by kumar8074** [comment]
Hey @abhisheks008 can i cjange the dataset used in this project? the dataset that have been previously used is heavily imbalanced

**Comment by abhisheks008** [comment]
> Hey @abhisheks008 can i change the dataset used in this project? the dataset that have been previously used is heavily imbalanced

Can you share the dataset URL/source you are planning to use here?

**Comment by kumar8074** [comment]
https://www.kaggle.com/datasets/architsharma01/loan-approval-prediction-dataset

**Comment by abhisheks008** [comment]
Cool go ahead @kumar8074

**Comment by kumar8074** [comment]
Hey @abhisheks008 i'm almost done with the project and accuracy is ticking around 95% i know it's slightly lesser than the previous one but the previous model is not actually accuracte because the model has `2 neurons` in the output layer but for the binary classification it should be `1` also it uses `softmax activation` for the output layer but the recommended for binary classification is `sigmoid` also it uses the 'categorical_crossentropy' as the loss function but the recommended is 'binary_crossentropy'.

However if you want me to improve the evaluation scores more i can do that by hyperparameter tuning which will take couple of more days need to run some more experiments

**Comment by abhisheks008** [comment]
> Hey @abhisheks008 i'm almost done with the project and accuracy is ticking around 95% i know it's slightly lesser than the previous one but the previous model is not actually accuracte because the model has `2 neurons` in the output layer but for the binary classification it should be `1` also it uses `softmax activation` for the output layer but the recommended for binary classification is `sigmoid` also it uses the `categorical_crossentropy` as the loss function but the recommended is `binary_crossentropy`.
> 
> However if you want me to improve the evaluation scores more i can do that by hyperparameter tuning which will take couple of more days need to run some more experiments

It's better to increase the accuracy a bit. You can take your time and come up with better solution.

**Comment by kumar8074** [comment]
> > Hey @abhisheks008 i'm almost done with the project and accuracy is ticking around 95% i know it's slightly lesser than the previous one but the previous model is not actually accuracte because the model has `2 neurons` in the output layer but for the binary classification it should be `1` also it uses `softmax activation` for the output layer but the recommended for binary classification is `sigmoid` also it uses the `categorical_crossentropy` as the loss function but the recommended is `binary_crossentropy`.
> > 
> > However if you want me to improve the evaluation scores more i can do that by hyperparameter tuning which will take couple of more days need to run some more experiments
> 
> It's better to increase the accuracy a bit. You can take your time and come up with better solution.

Sure Done 👍

**Comment by kumar8074** [comment]
Hey @abhisheks008 I'm done with the project.
I've forked your repository earlier so for making pull request should i just replace the files under this project folder and make pull request  using the entire forked repository ?? or should i just create a seperate folder and make a pull request for that folder only ?? what should i do??

**Comment by abhisheks008** [comment]
Make a PR for your codes only.

**Comment by abhisheks008** [comment]
This issue is not going to work.
